protocol/x/clob/keeper/msg_server_cancel_orders.go

```diff
@@ -22,6 +26,20 @@ func (k msgServer) CancelOrder(
 
 	defer func() {
 		if err != nil {
+			// Gracefully handle the case where the order was already removed from state.
```

Comment:
Suggestion: update this comment to explain why this happens

Also add a TODO here with a ticket to fix this for most cases (can use this ticket: https://linear.app/dydx/issue/CLOB-778/write-order-removals-to-state-store-uncommitted-transient-store). Ideally we also create a low priority M6 ticket to fix this for fully-filled orders (likely requires `PrepareProposal` changes).

Additionally, can we stat how often this happens?